### PR TITLE
TT-7903 DX-661 added otel in distributed tracing section

### PR DESCRIPTION
**Pull request by sredxny**
DX Ticket: https://tyktech.atlassian.net/browse/DX-661
---
### Preview Link
https://deploy-preview-3186--tyk-docs.netlify.app/docs/nightly/advanced-configuration/opentracing/

### Description
As we are about to release otel in tyk v5.2 we need to expose documentation of this new feature, how to enable, the config options and which data is added to the spans

Jira link: https://tyktech.atlassian.net/browse/TT-7903

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labelled** the PR

### CODE REVIEW

File: tyk-docs/content/advanced-configuration/opentracing.md [hunk 1]
```diff
@@ -8,14 +8,71 @@ menu:
...
+
+#### Advanced OpenTelemetry Capabilities
+##### Context Propagation and Sampling
+Tyk supports context propagation and configurable sampling strategies ("AlwaysOn", "AlwaysOff", "TraceIDRatioBased").
+
+#### Configuring Connection to OpenTelemetry Backend
+Choose between HTTP and gRPC for the backend connection by configuring the `exporter` field to "http" or "grpc".
+
+#### Further Configuration Details
+For more in-depth information on the configuration options available, please refer to our [OpenTelemetry Configuration Details Page](https://tyk.io/docs/tyk-oss-gateway/configuration/#opentelemetry).
```
Comment (sredxny) [inline tyk-docs/content/advanced-configuration/opentracing.md:73]: This links should be updated once we re-generate and release the config options of otel

File: tyk-docs/content/advanced-configuration/opentracing.md [hunk 2]
```diff
@@ -8,14 +8,71 @@ menu:
...
+"opentelemetry": {
+  "enabled": true
+  }
+}
+```
+
+Alternatively, you can use the equivalent environment variable to enable OpenTelemetry.
+
+##### Step 2: Enable at API Level
+After enabling OpenTelemetry at the gateway level, activate it for the specific APIs you want to monitor. Edit the respective API definition to set `detailed_tracing` to `true`.
```
Comment (tbuchaillot) [inline tyk-docs/content/advanced-configuration/opentracing.md:40]: What `detailed_tracing ` do? How does an API trace looks with and without detailed_tracing enabled?

File: tyk-docs/content/advanced-configuration/opentracing.md [hunk 3]
```diff
@@ -8,14 +8,71 @@ menu:
...
+
+#### Customizing Tracing
```
Comment (tbuchaillot) [inline tyk-docs/content/advanced-configuration/opentracing.md:42]: What's the customizable part of this?

File: tyk-docs/content/advanced-configuration/opentracing.md [hunk 4]
```diff
@@ -8,14 +8,71 @@ menu:
...
+##### Tyk-Specific Attributes and Tags
+Tyk sets specific span attributes automatically:
+
+- `tyk.api.name`: API name
+- `tyk.api.orgid`: Organization ID
+- `tyk.api.id`: API ID
+- `tyk.api.path`: API listen path
```
Comment (tbuchaillot) [inline tyk-docs/content/advanced-configuration/opentracing.md:49]: Please add as well resource attributes

File: tyk-docs/content/advanced-configuration/opentracing.md [hunk 5]
```diff
@@ -8,14 +8,71 @@ menu:
...
+- `http.method`: HTTP request method.
+- `http.scheme`: URL scheme.
+- `http.status_code`: HTTP response status code.
+- `http.url`: Full HTTP request URL.
+
+For the full list and details, refer to the official [OpenTelemetry Semantic Conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#common-attributes).
+
+#### Advanced OpenTelemetry Capabilities
+##### Context Propagation and Sampling
+Tyk supports context propagation and configurable sampling strategies ("AlwaysOn", "AlwaysOff", "TraceIDRatioBased").
```
Comment (tbuchaillot) [inline tyk-docs/content/advanced-configuration/opentracing.md:67]: How can you set different ratios? How does this affect the traces? More info

File: tyk-docs/content/advanced-configuration/opentracing.md [hunk 6]
```diff
@@ -8,14 +8,71 @@ menu:
...
+
+```json
+{
+"opentelemetry": {
+  "enabled": true
+  }
+}
+```
+
+Alternatively, you can use the equivalent environment variable to enable OpenTelemetry.
```
Comment (dcs3spp) [inline tyk-docs/content/advanced-configuration/opentracing.md:37]: Name and link the equivalent environment variable here

File: tyk-docs/content/advanced-configuration/opentracing.md (hunk 2 above)
Comment (dcs3spp) [reply tyk-docs/content/advanced-configuration/opentracing.md:40]: similar to comment above, what kind of information will detailed tracing log and maybe include an example of an API trace.

File: tyk-docs/content/advanced-configuration/opentracing.md [hunk 7]
```diff
@@ -8,14 +8,71 @@ menu:
...
+
+##### Step 2: Enable at API Level
+After enabling OpenTelemetry at the gateway level, activate it for the specific APIs you want to monitor. Edit the respective API definition to set `detailed_tracing` to `true`.
+
+#### Customizing Tracing
+##### Tyk-Specific Attributes and Tags
+Tyk sets specific span attributes automatically:
+
+- `tyk.api.name`: API name
```
Comment (dcs3spp) [inline tyk-docs/content/advanced-configuration/opentracing.md:46]: For consistent styling with other bullet points end the bullet points below with .
```suggestion
- `tyk.api.name`: API name.
```

File: tyk-docs/content/advanced-configuration/opentracing.md [hunk 8]
```diff
@@ -8,14 +8,71 @@ menu:
...
+- `tyk.api.orgid`: Organization ID
+- `tyk.api.id`: API ID
+- `tyk.api.path`: API listen path
+
+When tagging is enabled in the API definition, tags are added as another span attribute, keyed as `tyk.api.tags`.
```
Comment (dcs3spp) [inline tyk-docs/content/advanced-configuration/opentracing.md:51]:
```suggestion
When tagging is enabled in the API definition, tags are added as another span attribute. These are keyed as `tyk.api.tags`.
```

**Review by dcs3spp — CHANGES_REQUESTED** [review]
PR looks good to me @sredxny. I made some minor suggestions. Once review comments have been addressed we can merge the PR.

**Comment by sredxny** [comment]
@tbuchaillot @dcs3spp lets close this, and review https://github.com/TykTechnologies/tyk-docs/pull/3212

There I addressed all your comments, and also contains the new structure of the distributed tracing section

**Comment by sredxny** [comment]
closing in favor of: https://github.com/TykTechnologies/tyk-docs/pull/3212

**Pull request closed without merging**